src/backend/src/lib.rs

```diff
@@ -154,36 +155,57 @@ pub struct Config {
     pub ecdsa_key_name: String,
     // A list of allowed callers to restrict access to endpoints that do not particularly check or use the caller()
     pub allowed_callers: Vec<Principal>,
+    pub supported_credentials: Option<Vec<SupportedCredential>>,
+    /// Root of trust for checking canister signatures.
+    pub ic_root_key_raw: Option<Vec<u8>>,
+}
+
+fn set_config(arg: InitArg) {
+    let InitArg {
+        ecdsa_key_name,
+        allowed_callers,
+        supported_credentials,
+        ic_root_key_der,
+    } = arg;
+    mutate_state(|state| {
+        let ic_root_key_raw = match extract_raw_root_pk_from_der(
+            &ic_root_key_der.unwrap_or_else(|| IC_ROOT_PK_DER.to_vec()),
+        ) {
+            Ok(root_key) => root_key,
+            Err(msg) => panic!("{}", format!("Error parsing root key: {msg}")),
+        };
+        state
+            .config
+            .set(Some(Candid(Config {
+                ecdsa_key_name,
+                allowed_callers,
+                supported_credentials,
+                ic_root_key_raw: Some(ic_root_key_raw),
+            })))
+            .expect("setting config should succeed");
```

Comment:
If I read correctly, `.set()` can fail only if the data is too big to fit into memory.  The memory is expanded if necessary, so it tentatively looks to me as if the expect could be `.unwrap_or_else(|_| unreachable!("This data structure is too small to exhaust stable memory."))`?  I guess the rest of memory could be filled with accounts, and this is taking just the last few bytes so then I guess it could happen.  OK!  Please shout if I missed something but at the moment `expect` looks correct to me.

Reply:
Better be on the safe side.